src/main/java/no/unit/nva/publication/MainHandler.java

```diff
@@ -0,0 +1,162 @@
+        this.apiScheme = environment.get(API_SCHEME_ENV)
+                .orElseThrow(() -> new IllegalStateException(ENVIRONMENT_VARIABLE_NOT_SET + API_SCHEME_ENV));
+    }
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        String authorization;
+        UUID identifier;
+        try {
+            JsonNode event = objectMapper.readTree(input);
+            authorization = Optional.ofNullable(event.at(HEADERS_AUTHORIZATION).textValue())
+                    .orElseThrow(() -> new IllegalArgumentException(MISSING_AUTHORIZATION_IN_HEADERS));
+            identifier = UUID.fromString(Optional.ofNullable(event.at(PATH_PARAMETERS_IDENTIFIER).textValue())
+                    .orElseThrow(() -> new IllegalArgumentException(MISSING_IDENTIFIER_IN_PATH_PARAMETERS)));
+        } catch (Exception e) {
+            e.printStackTrace();
+            objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
+                    Problem.valueOf(BAD_REQUEST, e.getMessage())), headers(), SC_BAD_REQUEST));
+            return;
+        }
+
+        log("Request for identifier: " + identifier.toString());
+
+        try {
+            URI apiUrl = new URIBuilder()
+                    .setScheme(apiScheme)
+                    .setHost(apiHost)
+                    .build();
+
+            JsonNode resource = resourcePersistenceService.fetchResource(identifier, apiUrl.toString(), authorization);
+            JsonNode publication = resource.at(ITEMS_0);
+
+            if (publication.isMissingNode()) {
+                objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
+                        Problem.valueOf(NOT_FOUND)), headers(), SC_NOT_FOUND));
+                return;
+            }
+
+            ContextUtil.injectContext(publication, getPublicationContext());
+
+            objectMapper.writeValue(output, new GatewayResponse<>(
+                    objectMapper.writeValueAsString(publication), headers(), SC_OK));
+        } catch (ProcessingException | WebApplicationException e) {
+            e.printStackTrace();
+            objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
+                    Problem.valueOf(BAD_GATEWAY, e.getMessage())), headers(), SC_BAD_GATEWAY));
+        } catch (Exception e) {
+            e.printStackTrace();
+            objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
+                    Problem.valueOf(INTERNAL_SERVER_ERROR, e.getMessage())), headers(), SC_INTERNAL_SERVER_ERROR));
+        }
+    }
+
+    private JsonNode getPublicationContext() throws IOException {
+        return objectMapper.readTree(
+                Thread.currentThread().getContextClassLoader().getResourceAsStream(PUBLICATION_CONTEXT_JSON));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by ObjectMapper.readTree. Either a) close the object returned by *getResourceAsStream()* in a try-finally block or b) close the resource by declaring the object returned by *getResourceAsStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).